### ISSUE TITLE
HardNested cracking taking too long

### ISSUE THREAD

**Issue by Luk164** [AloneLiberty/FlipperNestedRecovery#5]
Cracking my .nonces was taking way too long. I was instructed to create a new issue for this.

I am attaching the .nonces file:
[.nested.zip](https://github.com/AloneLiberty/FlipperNestedRecovery/files/11361423/default.nested.zip)

**Comment by AloneLiberty** [comment]
Can you give more info about your PC and problematic sector?
I've successfully recovered all keys on Linux (x86) 
[312CFE0B.keys.zip](https://github.com/AloneLiberty/FlipperNestedRecovery/files/11361589/312CFE0B.keys.zip)

**Comment by AloneLiberty** [comment]
Missed one sector:
```
Recovering key type B, sector 30
[=] Hardnested attack starting...
[=] ---------+---------+---------------------------------------------------------+-----------------+-------
[=]          |         |                                                         | Expected to brute force
[=]  Time    | #nonces | Activity                                                | #states         | time 
[=] ---------+---------+---------------------------------------------------------+-----------------+-------
[=]        0 |       0 | Start using 32 threads                                  |                 |
[=]        0 |       0 | Brute force benchmark: 1817 million (2^30,8) keys/s     | 140737488355328 |   22h
[=]        0 |       0 | Using 235 precalculated bitflip state tables            | 140737488355328 |   22h
[=]        6 |     256 | Loading nonces from file                                |               0 |    0s
[=]       11 |     512 | Loading nonces from file                                |               0 |    0s
[=]       14 |     768 | Loading nonces from file                                |               0 |    0s
[=]       14 |    1024 | Loading nonces from file                                |               0 |    0s
[=]       15 |    1280 | Loading nonces from file                                |               0 |    0s
[-] No match for the First_Byte_Sum (127), is the card a genuine MFC Ev1? 
Found 0 key(s): []
```
You have problems with it?

**Comment by Luk164** [comment]
Hopefully it is just an unused sector. Could you share the command you used to generate all the keys? I only managed to figure out how to do it one by one

**Comment by Luk164** [comment]
What is also interesting is that we have nonces for 22-39 (1-22 were covered by default key database) but F0 reports 40 sectors. So nonces 40A and 40B were not captured. Along with the ?corrupted? 30B this could warrant further research.

**Comment by AloneLiberty** [comment]
There are 40 sectors, but they are counting from zero (so last sector is 39).
About 30 sector - you should try collecting nonces again, probably it will help
Back to our issue: what is your OS and CPU?

> Could you share the command you used to generate all the keys?

Download latest release (2.1.2) and just point to manifest file (UID.nonces)

**Comment by Luk164** [comment]
My os is Win11 Education 22H2 build 22621.1555
My cpu is Intel(R) Core(TM) i7-9750H

Will try to read the card again later today and post the result

**Comment by AloneLiberty** [comment]
And you are using native build (not MSYS/Cygwin/etc)?

**Comment by Luk164** [comment]
I am running via Conda but I would consider that as native.
Conda version 23.3.1 (miniconda)
Python version 3.10.10.final.0

BTW Repeating the scan helped, I have all the keys now :)

**Comment by AloneLiberty** [comment]
Issue should be partially solved with 1.3.0 release of FlipperNested (will check for first_byte_sum)

**Comment by armdn** [comment]
Same problem, nested attached.
[nested.zip](https://github.com/AloneLiberty/FlipperNestedRecovery/files/11372731/nested.zip)

**Comment by AloneLiberty** [comment]
@armdn Which sector cause error in your case?
UPD: Found it, your case is different, will investigate it 
UPD 2: Issue fixed, working on 000000000000 false positives

**Comment by AloneLiberty** [comment]
@armdn "Fixed" your issues in 2.3.0, can you retest when it will compile (in ~30 minutes) or compile yourself?

**Comment by Ladish-Ark** [comment]
Having the same issue here I think. I have attached them.
[Nested.zip](https://github.com/AloneLiberty/FlipperNestedRecovery/files/11520750/Nested.zip)

**Comment by AloneLiberty** [comment]
@Callum all keys recovered, probably reason for deleting the message.
[84DED0E7.keys.txt](https://github.com/AloneLiberty/FlipperNestedRecovery/files/11521851/84DED0E7.keys.txt)

Closing as fixed due to no bugs for 2 weeks

**Comment by armdn** [comment]
> @armdn "Fixed" your issues in 2.3.0, can you retest when it will compile (in ~30 minutes) or compile yourself?

Ok, will do the test in next few days.
